drivers/scheduler/k8s/k8s.go

```diff
@@ -5113,11 +5134,89 @@ func (k *K8s) CreateCsiSnapsAndValidate(ctx *scheduler.Context, snapClass string
+	// This test will validate the content of the snapshot as opposed to just verify creation of snapshot.
+	timestamp := request.Timestamp
+	namespace := request.Namespace + timestamp
+	clientset, err := getKubeClient("")
+	if err != nil {
+		logrus.Errorf("Failed to get kube client for clone test: %s", err)
+		return err
+	}
+
+	storageClassName := request.StorageclassName
+	// for FADA test, find a FA SC on our own that has pure_block param, if no FA SC, skip the test
+	if request.FADA {
+		fadaStorageClassName := ""
+		SCs, err:= clientset.StorageV1().StorageClasses().List(context.TODO(), metav1.ListOptions{})
+		if err != nil {
+			logrus.Errorf("Failed to get list of SCs from cluster")
+			return err
+		}
+
+		for _, SC := range SCs.Items {
+			if SC.Parameters["backend"] == PureBlock {
+				fadaStorageClassName = SC.Name
+			}
+		}
+
+		if fadaStorageClassName == "" {
+			logrus.Warn("No FADA SC on the cluster, skipping the test")
+			return nil
+		}
+		storageClassName = fadaStorageClassName
+	}
+	logrus.Infof("Procedding with SC %s", storageClassName)
+
+	// create a namespace for snapshot and restore test
+	nsSpec := &v1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}}
+	_, err = k8sCore.CreateNamespace(nsSpec)
+	if err != nil {
+		logrus.Errorf("Failed to create new namespace %s for clone test: %s", namespace, err)
+		return err
+	}
+
+	originalPVCName := request.OriginalPVCName
+	restoredPVCName :=  request.RestoredPVCName+ timestamp
+	dirtyData := "thisIsJustSomeRandomText"
+
+	logrus.Infof("Create original PVC: %s", originalPVCName)
+	originalPVCSpec := MakePVC(namespace, originalPVCName, request.StorageclassName)
+	originalPVC, err := k8sCore.CreatePersistentVolumeClaim(originalPVCSpec)
+
+	originalPodSpec := MakePod(originalPVCSpec.Namespace, []*v1.PersistentVolumeClaim{originalPVC}, "ls", false)
+
+	originalPod, err := clientset.CoreV1().Pods(namespace).Create(context.TODO(), originalPodSpec, metav1.CreateOptions{})
+	if err != nil {
+		logrus.Errorf("Failed to create original pod for clone: err")
+		return err
+	}
+
+	if err = k.waitForPodToBeReady(originalPod.Name, originalPod.Namespace); err != nil {
+		return &scheduler.ErrFailedToSchedulePod{
+			App: nil,
+			Cause:   fmt.Sprintf("Original pod is not ready. Error: %v", err),
+		}
+	}
+	fmt.Println("Original pod up and running, creating multiple PVC by cloning")
+	for i :=0; i < 3; i++ {
+		data := fmt.Sprint(dirtyData, strconv.Itoa(int(time.Now().Unix())))
+		err = k.cloneAndVerify(data, namespace, request.StorageclassName, fmt.Sprint(restoredPVCName, i), originalPod.Name, originalPVC.Name)
+		if err != nil {
+			logrus.Errorf("failed to validate cloned PVC content: %s ", err)
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (k *K8s) createSnapshotAndVerify(data, snapName, namespace, storageClass, snapClass, restoredPVCName, originalPod, originalPVC string) error {
```

Comment:
To be analogy with cloneAndVerify function, change the function name to be `snapshotAndVerify`